### ISSUE TITLE
New rules created during prerelease cycle appear in rules list, but rule doc not available (HTTP 404)

### ISSUE THREAD

**Issue by platinumazure** [eslint/eslint-release#22]
If a rule is created during a prerelease cycle and then the prerelease script is run, the rules list is updated with the new rule but the rule's docs are not available (files are only copied into, e.g., docs/5.0.0/rules instead of docs/rules).

Presumably, the prerelease script should be more consistent and either:

* Not include prerelease rules on the rules list (i.e., rules list and rule doc pages are from latest stable release)
* Include prerelease rules and also copy docs to docs/rules

I lean towards the latter because rule pages will include info on when the rule was introduced, so it should be easy to spot rules that were introduced during the prerelease.

**Comment by revolter** [comment]
And should it also display a warning that it's not a released rule yet?

**Comment by platinumazure** [comment]
@revolter Good call, although I think that will take more work. I've created eslint/eslint#10412, which would be a prerequisite. Thanks!

In this issue, I'd rather focus on the bugfix of ensuring all rule docs are copied appropriately to `docs/rules` during site generation.
